Split the docs for "Standard Library" into "Base" and "Standard Library"

KristofferC opened the pull request:
The docs right now is split into Manual and Standard Library (and Developer Docs). However, we are now more and more differentiating between what is part of "Base" and what is the "Standard Library". Therefore, I think it makes sense to split the header that is currently called "Standard Library" into two parts; one called "Base" (corresponding to functionality in the `base` foler) and one called "Standard Library" (which corresponds to the modules in `stdlib`).

WIP because the introduction to Base needs to be rewritten to not call it the Standard Library, and because there should probably be an introduction corresponding to the Standard Library header.

KristofferC committed:
split out a part of stdlib into a Base category in docs

StefanKarpinski commented:
💯

Sacha0 commented:
This pull request brings the discussion in https://github.com/JuliaLang/julia/pull/23876#issuecomment-333163762 to mind. Migrating to `stdpkg`/"standard packages" would be a nice way to avoid ambiguity / confusion and maintain clear separation. Best!

KristofferC commented:
Personally, I find the Base / Standard Library distinction to work quite well. Standard packages sounds a bit like an oxymoron to me.

KristofferC commented:
Alright, updated this. For those places in the Base documentation that were referring to Base as "the standard library", I changed this to "Julia Base". Any other suggestions?

There are a few places in the Manual section (Date and Datetime, Profiling) that should probably be moved completely to their stdlib documentation, but this can be done later.

This is no longer WIP so please review.

fredrikekre reviewed (approved):
Very nice!

fredrikekre commented:
Timeout on one Appveyor worker.

JeffBezanson commented on doc/make.jl:
```diff
@@ -162,13 +139,13 @@ const PAGES = [
     build     = joinpath(pwd(), "_build/html/en"),
-    modules   = [Base, Core, BuildSysImg, DelimitedFiles, Test, Mmap, SharedArrays, Profile,
-                 Base64, FileWatching, Dates, IterativeEigensolvers, Unicode, Distributed, Printf],
+    modules   = [Base, Core, BuildSysImg, [Module(stdlib.stdlib) for stdlib in STDLIB_DOCS]...],
```
Does this work? I thought `Module(x)` made a new empty module.

Great change though :+1:

KristofferC replied:
> I thought `Module(x)` made a new empty module.

Well, clearly you haven't been programming Julia long enough...

It works in the sense that the docs render correctly as far as I can see.

JeffBezanson replied:
That's weird; something is up. `stdlib.stdlib` is a Symbol right?

```
julia> using Unicode

julia> Module(:Unicode) === Unicode
false
```

KristofferC replied:
Ok, it is wrong but the reason it still work, I think, is that none of the stdlib modules have any submodules, which is what Documenter uses this information. Will correct.
